docs/oas.yaml

```diff
@@ -0,0 +1,483 @@
+        in: path
+        description: The id of the material to update
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/MaterialBody'
+      responses:
+        "200":
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MaterialResponse'
+        "400":
+          description: Client error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "401":
+          description: Authentication error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "403":
+          description: Permission error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "404":
+          description: Material not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "500":
+          description: Server error
+    delete:
+      tags:
+      - materials
+      summary: Delete material by ID
+      description: Deletes a single material
+      operationId: deleteMaterialById
+      parameters:
+      - name: materialId
+        in: path
+        description: The id of the material to update
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+      responses:
+        "204":
+          description: Material deleted successfully
+        "404":
+          description: Material not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "500":
+          description: Server error
+  /materials/{materialId}/users:
+    get:
+      tags:
+      - materials
+      summary: Get users who purchased the material by ID
+      description: Get users who purchased the material by ID
+      operationId: getUsersForMaterial
+      parameters:
+      - name: materialId
+        in: path
+        description: The id of the material to get users who purchased it
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+      responses:
+        "200":
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+        "404":
+          description: Material not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /classes:
+    post:
```

Comment:
This post should be "/classes/courseId", because of what was discussed to link the classes with their corresponding course. I don't know if this will involve also modifying the class schema, to include the course id or something like that.

Reply:
Ok, yes, I change this to indicate that the course id of the class is passed as a parameter. I also add to the class schema a property for the course id.
Thanks Tomas!